cli/cli.ts

```diff
@@ -3451,19 +3457,47 @@ function publishGistCoreAsync(): Promise<void> {
             filesMap['pxt.json'] = {
                 "content": JSON.stringify(pxtConfig, null, 4)
             }
-            return pxt.github.publishGist(filesMap, pxtConfig.description, false)
+            console.log("Uploading....")
+            return pxt.github.publishGist(token, forceNewGist, filesMap, pxtConfig.description, pxtConfig.gistId, publishPublicly)
         })
         .then((published_id) => {
-            console.log(`Success, your gist url is https://gist.github.com/${published_id}`);
+            console.log(`Success, view your gist at https://gist.github.com/${published_id}`);
             console.log(`You can load your published project at ${pxt.appTarget.appTheme.homeUrl}#pub:gh/gists/${published_id}`)
+
+            // Save gist id to pxt.json
+            mainPkg.config.gistId = published_id;
```

Comment:
The gist id, should only be a local pxt thing. Whenever we publish to gist, I don't think we should send in the gistId. 
If we make sure to strip those out, then any fork has no attached gist id.